### PR TITLE
Fixes #22 C# gha workflow

### LINKED ISSUE #22: Set up continuous integration for C++ and C#
Windows and Linux

### PR DESCRIPTION
**Pull request by Felixmil**
Fixes #22

image

### CODE REVIEW

**Comment by rwmcintosh** [comment]
I think we have to be observant about this. I believe the tests are not very reliable, but let's start with this.

**Comment by rwmcintosh** [comment]
Fixes #22

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -1,10 +1,11 @@
 # Workflow derived from https://github.com/r-lib/actions/tree/v2/examples
 # Need help debugging build failures? Start at https://github.com/r-lib/actions#where-to-find-help
 on:
-  push:
-    branches: [main, master, develop]
-  pull_request:
-    branches: [main, master, develop]
+  # push:
```
Comment (Felixmil) [inline .github/workflows/R-CMD-check.yaml:4]: Please uncomment, R CMD Check should be ran on each commit

File: .github/workflows/pkgdown.yaml [hunk 2]
```diff
@@ -1,12 +1,12 @@
 # Workflow derived from https://github.com/r-lib/actions/tree/v2/examples
 # Need help debugging build failures? Start at https://github.com/r-lib/actions#where-to-find-help
 on:
-  push:
-    branches: [main, master, develop]
-  pull_request:
-    branches: [main, master, develop]
-  release:
-    types: [published]
+  # push:
```
Comment (Felixmil) [inline .github/workflows/pkgdown.yaml:4]: same here: uncomment.

File: .github/workflows/test-coverage.yaml [hunk 3]
```diff
@@ -1,10 +1,11 @@
 # Workflow derived from https://github.com/r-lib/actions/tree/v2/examples
 # Need help debugging build failures? Start at https://github.com/r-lib/actions#where-to-find-help
 on:
-  push:
-    branches: [main, master, develop]
-  pull_request:
-    branches: [main, master, develop]
+  # push:
```
Comment (Felixmil) [inline .github/workflows/test-coverage.yaml:4]: Same here: uncomment.

**Comment by Felixmil** [comment]
R actions should start only after binaries successfully built.

**Comment by Felixmil** [comment]
@rwmcintosh The `.so` file is apparently not commited during the Linux step: https://github.com/Open-Systems-Pharmacology/rSharp/actions/runs/9059706133/job/24887856458#step:8:28

However the `.dll` files are commited during the Windows build: https://github.com/Open-Systems-Pharmacology/rSharp/actions/runs/9059706133/job/24887924340#step:10:32

**Comment by rwmcintosh** [comment]
> @rwmcintosh The `.so` file is apparently not commited during the Linux step: https://github.com/Open-Systems-Pharmacology/rSharp/actions/runs/9059706133/job/24887856458#step:8:28
> 
> However the `.dll` files are commited during the Windows build: https://github.com/Open-Systems-Pharmacology/rSharp/actions/runs/9059706133/job/24887924340#step:10:32

It could be because gcc builds the binary equivalent when source is unchanged where msvc does not. I will commit a change in the C++ cod to test it today, but then I need to change it back again before merging.

**Comment by rwmcintosh** [comment]
> > @rwmcintosh The `.so` file is apparently not commited during the Linux step: https://github.com/Open-Systems-Pharmacology/rSharp/actions/runs/9059706133/job/24887856458#step:8:28
> > However the `.dll` files are commited during the Windows build: https://github.com/Open-Systems-Pharmacology/rSharp/actions/runs/9059706133/job/24887924340#step:10:32
> 
> It could be because gcc builds the binary equivalent when source is unchanged where msvc does not. I will commit a change in the C++ cod to test it today, but then I need to change it back again before merging.

@Felixmil confirmed the above. Changes in cpp source will trigger a commit of .so. All Linux and Windows build/commit is complete.

**Comment by Felixmil** [comment]
C# binaries are now built by github actions and pushed to repository (in PR branches, before merge)

@PavelBal Can we merge ?

**Comment by PavelBal** [comment]
Let's try out.

**Review by PavelBal — APPROVED** [review]

**Pull request merged**